rclcpp/include/rclcpp/qos.hpp

```diff
@@ -90,18 +90,36 @@ struct RCLCPP_PUBLIC KeepLast : public rclcpp::QoSInitialization
 };
 
 /// Encapsulation of Quality of Service settings.
+/**
+ * Quality of Service settings contains parameters for the underlying middleware, typically DDS,
+ * to tune communication between nodes. See
+ * <a href="https://index.ros.org/doc/ros2/Concepts/About-Quality-of-Service-Settings/">
+ About Quality of Service Settings</a> for an explanation.
+ */
 class RCLCPP_PUBLIC QoS
 {
 public:
-  /// Constructor which allows you to construct a QoS by giving the only required settings.
+  /// Constructor which allows creating a QoS by specifying only the history policy and history depth.
+  /**
+   * When using the default initial profile, this uses \link rclcpp::ReliabilityPolicy::Reliable
+   * ReliabilityPolicy::Reliable\endlink and \link rclcpp::DurabilityPolicy::Volatile
+   * DurabilityPolicy::Volatile\endlink. See the global variable rmw_qos_profile_default for a full list.
+   *
+   * \param[in] qos_initialization Specifies history policy and history depth.
+   * \param[in] initial_profile Specifies other settings.
+   */
```

Comment:
```suggestion
  /// Constructor which allows creating a QoS by specifying only the history policy and history depth.
  /**
   * When using the default initial profile, the defaults will include:
   *
   *   - \link rclcpp::ReliabilityPolicy::Reliable ReliabilityPolicy::Reliable\endlink
   *   - \link rclcpp::DurabilityPolicy::Volatile DurabilityPolicy::Volatile\endlink
   *
   * See rmw_qos_profile_default for a full list of default settings.
   * If some other rmw_qos_profile_t is passed to initial_profile, then the defaults will derive from
   * that profile instead.
   *
   * \param[in] qos_initialization Specifies history policy and history depth.
   * \param[in] initial_profile The rmw_qos_profile_t instance on which to base the default settings.
   */
```

Just cleaning it up a bit.

Whenever possible we should try to avoid leaning on rmw or rcl in the rclcpp documentation. Ideally rclcpp's documentation would be self describing, but in some cases like this referring to rmw is counter productive.